### PR TITLE
Reduce repetition by creating fieldset component

### PR DESCRIPTION
**Pull request by stefpiatek**
Created wrapper component to handle common field use, also can rely that if validation works once then it'll work overall so could remove testing the validation in the add report form 🎉

### CODE REVIEW

File: src/components/reports/FieldSet.test.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+     * When the form field is touched and then blurred
+     * Then there should be a validation error
+     */
+    test('Form validation is preserved', async () => {
+      // Arrange
+      render(<TestComponent/>);
+
+      // Act
+      const testField = screen.getByLabelText(/test/i);
+      fireEvent.touchStart(testField)
```
Comment (jhughes982) [inline src/components/reports/FieldSet.test.tsx:79]: Any particular reason why a touch event is used here? Is this how users are expected to interact with the form?

**Review by jhughes982 — APPROVED** [review]
Looks good to approve with just a few comments for possible improvement if applicable.

Just as a possible thought point. The only thing to be cautious of is testing the functionality of an external library that we can reasonably expect to already work. However, since you have added your own abstraction with your FieldSet component it makes sense to verify that it still works in this case.

File: src/components/reports/ReportForm.tsx [hunk 2]
```diff
@@ -84,117 +85,51 @@ const ReportForm = () => {
...
-          <Field id="postCode" name="address.postCode"/>
-          <ErrorMessage component="p"  className="error-text" name="address.postCode"/>
-          <label htmlFor="country">Country</label>
-          <Field id="country" name="address.country"/>
-          <ErrorMessage component="p"  className="error-text" name="address.country"/>
+
+          <FieldSet as="textarea" name="address.streetAddress" label="Street Address"/>
+          <FieldSet name="address.city" label="City"/>
+          <FieldSet name="address.postCode" label="Post Code"/>
+          <FieldSet name="address.country" label="Country"/>
```
[the remark this answers is not in the document]
Comment (stefpiatek) [reply src/components/reports/ReportForm.tsx:92]: I think that's a good call, I'll add it as an issue to think about after we're sure that we've fully defined the input data.

File: src/components/reports/FieldSet.test.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     * When a value is submitted
+     * Then the form should successfully submit
+     */
+    test('Form values are accessible through FieldSet wrapper', async () => {
+      // Arrange
+      render(<TestComponent/>);
+
+      // Act
+      const expected = "testValue";
+      fireEvent.change(screen.getByLabelText(/test/i), {
```
[the remark this answers is not in the document]
Comment (stefpiatek) [reply src/components/reports/FieldSet.test.tsx:56]: ah yes that makes sense, nah I think `userEvent` is more appropriate

File: src/components/reports/ReportForm.tsx [hunk 4]
```diff
@@ -84,117 +85,51 @@ const ReportForm = () => {
...
-          <Field id="city" name="address.city"/>
-          <ErrorMessage component="p"  className="error-text" name="address.city"/>
-          <label htmlFor="postCode">Post Code</label>
-          <Field id="postCode" name="address.postCode"/>
-          <ErrorMessage component="p"  className="error-text" name="address.postCode"/>
-          <label htmlFor="country">Country</label>
-          <Field id="country" name="address.country"/>
-          <ErrorMessage component="p"  className="error-text" name="address.country"/>
+
+          <FieldSet as="textarea" name="address.streetAddress" label="Street Address"/>
```
[the remark this answers is not in the document]
Comment (stefpiatek) [reply src/components/reports/ReportForm.tsx:89]: Such a good question, I think I'd need to pass through children for that, or pass through specific props which then map children into the FieldSet JSX

File: src/components/reports/FieldSet.test.tsx (hunk 1 above)
Comment (stefpiatek) [reply src/components/reports/FieldSet.test.tsx:79]: For this I wanted to make sure that it was simulating that they clicked on the field, but then moved away from it without filling in the required value - I don't think blur would work on it's own?

**Pull request merged**